src/main/java/org/killbill/billing/plugin/adyen/client/payment/service/checkout/CheckoutPaymentsResult.java

```diff
@@ -112,16 +117,23 @@ public PurchaseResult toPurchaseResult() {
             additionalData.put(AdyenPaymentPluginApi.PROPERTY_MERCHANT_ACCOUNT_CODE, merchantAccount);
         }
 
-        if(formParameter != null) {
+        if(paymentInfo.getIdentifierMap() != null && !paymentInfo.getIdentifierMap().isEmpty()) {
+            try {
+                String hashedIdStr = mapper.writeValueAsString(paymentInfo.getIdentifierMap());
+                additionalData.put("identifierMap", hashedIdStr);
+            } catch (JsonProcessingException ex) {
+                logger.error("Failed to save hashIdentifiers: " + ex.getMessage());
+            }
+        }
+        if(formParameter != null && !formParameter.isEmpty()) {
             try {
                 String formParams = mapper.writeValueAsString(formParameter);
                 additionalData.put("formParameter", formParams);
             } catch (JsonProcessingException ex) {
                 logger.error("Failed to save formParameters: " + ex.getMessage());
             }
         }
-
-        if(authResultKeys != null) {
+        if(authResultKeys != null && !authResultKeys.isEmpty()) {
```

Comment:
can we use CollectionUtils.isNotEmpty()?

Reply:
Same as above.